deploy/kube-config/influxdb/grafana-image/influxdb_service_discovery.go

```diff
@@ -0,0 +1,76 @@
+// Copyright 2014 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"fmt"
+	"strconv"
+	"time"
+
+	"github.com/golang/glog"
+	"k8s.io/kubernetes/pkg/api"
+	client "k8s.io/kubernetes/pkg/client/unversioned"
+)
+
+func main() {
+	glog.Info("influxdb service discovery")
+
+	c, err := client.NewInCluster()
+	if err != nil {
+		glog.Fatalf("Failed to make client: %v", err)
+	}
+
+	glog.Info("Looking up monitoring-influxdb service...")
+	var influxdb *api.Service
+	// Wait a bit longer for influxdb loadBalancer to be created
+	for t := time.Now(); time.Since(t) < 5*time.Minute; time.Sleep(10 * time.Second) {
+		influxdb, err = c.Services(api.NamespaceSystem).Get("monitoring-influxdb")
+		if err == nil && len(influxdb.Status.LoadBalancer.Ingress) > 0 {
+			break
+		} else if err == nil {
+			glog.Info("Service is up. Waiting for load balancer to be created...")
+		} else {
+			glog.Info("Waiting for service to come up...")
+		}
+	}
+	if influxdb == nil {
+		glog.Error("Failed to find the monitoring-influxdb service.")
+		return
+	}
+
+	var serviceIP string
+	if len(influxdb.Status.LoadBalancer.Ingress) > 0 {
+		ingress := influxdb.Status.LoadBalancer.Ingress[0]
+		serviceIP = ingress.IP
+		if serviceIP == "" {
+			serviceIP = ingress.Hostname
+		}
+		glog.Infof("Found monitoring-influxdb service with external IP: '%s'", serviceIP)
+	} else {
+		glog.Warning("Service monitoring-influxdb exists but there's no load balancer.")
```

Comment:
This is to support the "direct" access use case where InfluxDB service is exposed as "Load Balancer" type. Since Grafana continues to support this method as mentioned on http://docs.grafana.org/datasources/influxdb/, I thought it would be useful to have that support here as well.